perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,400 @@
+    :param min_rate_to_sleep: minimum rate needed to sleep until
+         it will be reset
+    :param from_archive: it tells whether to write/read the archive
+    :param archive: archive to store/retrieve items
+    :param ssl_verify: enable/disable SSL verification
+    """
+    RCHANNEL_MESSAGES = 'channels.messages'
+    RCHANNEL_INFO = 'channels.info'
+
+    HAUTH_TOKEN = 'X-Auth-Token'
+    HUSER_ID = 'X-User-Id'
+
+    PCHANNEL_NAME = 'roomName'
+    PCOUNT = "count"
+    POLDEST = "oldest"
+
+    def __init__(self, url=None, user_id=None, api_token=None, max_items=MAX_ITEMS,
+                 sleep_for_rate=False, min_rate_to_sleep=MIN_RATE_LIMIT,
+                 from_archive=False, archive=None, ssl_verify=True):
+
+        base_url = urijoin(url, API_EXTENSION)
+        self.user_id = user_id
+        self.api_token = api_token
+        self.max_items = max_items
+
+        super().__init__(base_url, archive=archive, from_archive=from_archive,
+                         ssl_verify=ssl_verify)
+        super().setup_rate_limit_handler(sleep_for_rate=sleep_for_rate, min_rate_to_sleep=min_rate_to_sleep)
+
+    def calculate_time_to_reset(self):
+        """Number of seconds to wait. They are contained in the rate limit reset header."""
+
+        time_to_reset = self.rate_limit_reset_ts - (datetime_utcnow().replace(microsecond=0).timestamp() + 1) * 1000
+        time_to_reset /= 1000
+
+        if time_to_reset < 0:
+            time_to_reset = 0
+
+        return time_to_reset
+
+    def channel_info(self, channel):
+        """Fetch information about a channel."""
+
+        params = {
+            self.PCHANNEL_NAME: channel,
+        }
+
+        path = urijoin(self.base_url, self.RCHANNEL_INFO)
+        response = self.fetch(path, params)
+
+        return response
+
+    def messages(self, channel, from_date, offset):
+        """Fetch messages from a channel."""
+
+        query = '{"_updatedAt": {"$gte": {"$date": "%s"}}}' % from_date.isoformat()
+
+        params = {
+            "roomName": channel,
+            "sort": '{"_updatedAt": 1}',
+            "count": self.max_items,
+            "offset": offset,
+            "query": query
+        }
+
+        path = urijoin(self.base_url, self.RCHANNEL_MESSAGES)
+        response = self.fetch(path, params)
+
+        return response
+
+    def fetch(self, url, payload=None, headers=None):
+        """Fetch the data from a given URL.
+
+        :param url: link to the resource
+        :param payload: payload of the request
+        :param headers: headers of the request
+
+        :returns a response object
+        """
+        headers = {
+            self.HAUTH_TOKEN: self.api_token,
+            self.HUSER_ID: self.user_id
+        }
+
+        logger.debug("RocketChat client message request with params: %s", str(payload))
+
+        if not self.from_archive:
+            self.sleep_for_rate_limit()
+
+        response = super().fetch(url, payload, headers=headers)
```

Comment:
Hi @valeriocos, Please have a look at this line.
In case a user provides a wrong `Channel Name` or `User Id`, an HTTP error is raised here. In that case, the log includes `error.response.reason` which is not that deductive. The actual cause for the error is returned in the body of the response. Hence I initially made the below change.
```
try:
      response = super().fetch(url, payload, headers=headers)
except requests.exceptions.HTTPError as error:
      logger.error(error.response.text)
      raise error
```

However, I think such a situation may happen in other APIs as well.(I found it in Pagure, Gitter, Groups.io when working with them) Hence, instead of addressing this in the backend a change in the core client should be addressed in separate PR. 
WDYT?

Reply:
Yes, I saw many error messages that were confusing in the backends.

Reply:
Thank you @lukaszgryglicki for the feedback. The PR https://github.com/chaoss/grimoirelab-perceval/pull/672 tries to fix the problem, but the solution is still not good.

It would be great if you can add to the corresponding issue (https://github.com/chaoss/grimoirelab-perceval/issues/671), some examples of error messages that are confusing for you. Thanks